src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/AuditConfig.java

```diff
@@ -0,0 +1,168 @@
+        boolean logRequestBody = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_LOG_REQUEST_BODY, true);
+        boolean resolveIndices = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_RESOLVE_INDICES, true);
+        boolean excludeSensitiveHeaders = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_EXCLUDE_SENSITIVE_HEADERS, true);
+
+        final List<String> defaultDisabledCategories = Arrays.asList(AuditCategory.AUTHENTICATED.toString(), AuditCategory.GRANTED_PRIVILEGES.toString());
+
+        EnumSet<AuditCategory> disabledRestCategories = AuditCategory.parse(getConfigList(
+                settings,
+                ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_REST_CATEGORIES,
+                defaultDisabledCategories));
+
+        EnumSet<AuditCategory> disabledTransportCategories = AuditCategory.parse(getConfigList(
+                settings,
+                ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_TRANSPORT_CATEGORIES,
+                defaultDisabledCategories));
+        final List<String> defaultIgnoredUsers = Arrays.asList("kibanaserver");
+
+        List<String> ignoredAuditUsers = getConfigList(
+                settings,
+                ConfigConstants.OPENDISTRO_SECURITY_AUDIT_IGNORE_USERS,
+                defaultIgnoredUsers);
+
+        List<String> ignoredComplianceUsersForRead = getConfigList(
+                settings,
+                ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_READ_IGNORE_USERS,
+                defaultIgnoredUsers);
+
+        List<String> ignoredComplianceUsersForWrite = getConfigList(
+                settings,
+                ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_IGNORE_USERS,
+                defaultIgnoredUsers);
+
+        List<String> ignoreAuditRequests = getConfigList(
+                settings,
+                ConfigConstants.OPENDISTRO_SECURITY_AUDIT_IGNORE_REQUESTS,
+                Collections.emptyList());
+
+
+        AuditConfig config = new AuditConfig(
+                restAuditingEnabled,
+                transportAuditingEnabled,
+                resolveBulkRequests,
+                logRequestBody,
+                resolveIndices,
+                excludeSensitiveHeaders,
+                ignoredAuditUsers,
+                ignoredComplianceUsersForRead,
+                ignoredComplianceUsersForWrite,
+                ignoreAuditRequests,
+                disabledRestCategories,
+                disabledTransportCategories);
+        return config;
+    }
+
+    private static List<String> getConfigList(final Settings settings,
+                                              final String key,
+                                              final List<String> defaultList) {
+        List<String> list = new ArrayList<>(settings.getAsList(key, defaultList));
+        if (list.size() == 1 && "NONE".equals(list.get(0))) {
```

Comment:
why are we allowing users to pass NONE? how is this different from an empty list?

Reply:
Good question. I retained what existed already.
YAML 
1) list: []
2) list: ["NONE"]  
3) list: "NONE" 

should all behave the same for now.
Only advantage I see is how users can explicitly say "NONE" instead of a list.

Reply:
Thanks for the explanation. We should determine if we want to support NONE as a valid option in the future. According to me an empty list should suffice. Will let @hardik-k-shah  and @dinusX chime in with their opinions.

Reply:
https://opendistro.github.io/for-elasticsearch-docs/docs/security-audit-logs/#exclude-categories 
We probably need the above if we allow users to set NONE. But yeah if we force the users to use [] then we wont need it.